Cython/Utility/MemoryView.pyx

```diff
@@ -514,7 +514,7 @@ cdef class memoryview(object):
         if flags & PyBUF_WRITABLE and self.view.readonly:
             raise ValueError("Cannot create writable memory view from read-only memoryview")
 
-        if flags & PyBUF_STRIDES:
+        if flags & PyBUF_ND:
```

Comment:
What happens in the strides case now?

Reply:
Oh, and it's also worth declaring the enum further up. :)

Reply:
Thanks for catching that. Fixed.

As for the strides case, `PyBUF_STRIDES` is a composite, which [includes `PyBUF_ND`]( https://github.com/python/cpython/blob/a757649fd6535c5e65481eb1077e30687421b92b/Include/object.h#L213 ). So this should catch both cases. Admittedly that is internal spec, so [this line from the docs]( https://docs.python.org/3/c-api/buffer.html#shape-strides-suboffsets ) may be more convincing...

> Note that each flag contains all bits of the flags below it.